### PR TITLE
Feature/do 4512 service acct role for local dev

### PR DESCRIPTION
**Pull request by ramnarayan-cofinityx**

### CODE REVIEW

**Comment by hgo-ds** [comment]
**Suggestion:**

- Consider adding a `ProjectRunConfigurationManager` like we already have for `dev` env. See https://github.com/Cofinity-X/cofinity-x-ba-traceability-foss/blob/dev/.run/Trace%20X%20with%20local%20IdP.run.xml as example.
- Replace all placeholders in that new RunConfiguration as concrete key value pairs in the configuration. 
- Drop all configuration which is declared as default in https://github.com/Cofinity-X/cofinity-x-ba-traceability-foss/blob/dev/tx-backend/src/main/resources/application.yml already.

**Comment by ramnarayan-cofinityx** [comment]
> **Suggestion:**
> 
> * Consider adding a `ProjectRunConfigurationManager` like we already have for `dev` env. See https://github.com/Cofinity-X/cofinity-x-ba-traceability-foss/blob/dev/.run/Trace%20X%20with%20local%20IdP.run.xml as example.
> * Replace all placeholders in that new RunConfiguration as concrete key value pairs in the configuration.
> * Drop all configuration which is declared as default in https://github.com/Cofinity-X/cofinity-x-ba-traceability-foss/blob/dev/tx-backend/src/main/resources/application.yml already.

@hgo-ds Implemented all the required changes.

File: .run/TraceX-Backend-localenv-profile.run.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      <env name="OAUTH2_CLIENT_ID" value="catena-client-id" />
+      <env name="OAUTH2_CLIENT_SECRET" value="r8jiU6MM5wCuaVL54mCLI6OeaWWRqzL4" />
+      <env name="OAUTH2_JWK_SET_URI" value="http://localhost:4011/realms/CX-Central/protocol/openid-connect/certs" />
+      <env name="OAUTH2_PROVIDER_TOKEN_URI" value="http://localhost:4011/realms/CX-Central/protocol/openid-connect/token" />
+      <env name="SPRING_DATASOURCE_PASSWORD" value="docker" />
+      <env name="SPRING_DATASOURCE_URL" value="jdbc:postgresql://localhost:5432/trace" />
+      <env name="SPRING_DATASOURCE_USERNAME" value="postgres" />
+      <env name="SPRING_PROFILES_ACTIVE" value="localenv" />
+      <env name="TRACEABILITY_BPN" value="BPNL000000000001" />
+      <env name="TRACEABILITY_URL" value="http://localhost:8080/" />
```
Comment (hgo-ds) [inline .run/TraceX-Backend-localenv-profile.run.xml]: **Suggestion:**
Consider to drop those lines which has no effect.

Comment (hgo-ds) [reply .run/TraceX-Backend-localenv-profile.run.xml]: Keep the file, just with as less env entries as possible:

```html
<component name="ProjectRunConfigurationManager">
  <configuration default="false" name="Trace X localenv with local IdP" type="SpringBootApplicationConfigurationType" factoryName="Spring Boot">
    <option name="ACTIVE_PROFILES" value="localenv" />
    <module name="tx-backend" />
    <option name="SPRING_BOOT_MAIN_CLASS" value="org.eclipse.tractusx.traceability.TraceabilityApplication" />
    <method v="2">
      <option name="Make" enabled="true" />
    </method>
  </configuration>
</component>
```

**Review by ds-asmierzchalski — DISMISSED** [review]
looks fine

**Pull request merged**